phoenix-core-client/src/main/java/org/apache/phoenix/compile/QueryCompiler.java

```diff
@@ -787,6 +790,33 @@ protected QueryPlan compileSingleFlatQuery(
         TableRef tableRef = context.getCurrentTable();
         PTable table = tableRef.getTable();
 
+        TTLExpression ttlExpr = table.getTTL();
```

Comment:
@comnetwork Looking for feedback if this is the right way to re-write the where clause by appending a boolean expression. In my testing it is working so far.